data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -100,18 +100,20 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
             throw new PatternSyntaxException("delete_value_regex is not a valid regex string", keyValueProcessorConfig.getDeleteValueRegex(), -1);
         }
 
-        if (keyValueProcessorConfig.getIncludeKeys() != null) {
-            if (keyValueProcessorConfig.getExcludeKeys() != null) {
-                if (keyValueProcessorConfig.getIncludeKeys().equals(keyValueProcessorConfig.getExcludeKeys())) {
-                    if (!keyValueProcessorConfig.getIncludeKeys().isEmpty()) {
-                        throw new IllegalStateException("Include keys and exclude keys set cannot be the same", null);
-                    }
-                }
-
-                excludeKeysSet.addAll(keyValueProcessorConfig.getExcludeKeys());
+        if (keyValueProcessorConfig.getIncludeKeys().equals(keyValueProcessorConfig.getExcludeKeys())) {
```

Comment:
I think you could absorb this into the validation logic that checks `Include keys and exclude keys set cannot have any overlap` since it is just a special case.